### DISCUSSION TITLE
Can you change auto-import locations?

### DISCUSSION THREAD

**Discussion by mezuzza** [microsoft/pyright#8612] [General]
Let's say we have the following directory contents:

```
foo/
  __init__.py
  _bar.py
```

The contents look like so:

```python
# __init__.py
from ._bar import baz

__all__ = ["baz"]
```

If I try to import auto-import baz, pyright seems to use `from foo._bar import baz` and I have to manually change the import. Is there any way to configure python, pyright, or anything else to have it default to `from foo import baz` without having to handle this manually?

For context, I'm using coc-pyright with vim8, so it's totally possible I'm missing something, but I've looked for a while and can't find a good solution to this

**Comment by erictraut** [comment]
I agree that it would be better if the auto-import feature didn't suggest private modules.

The auto-import logic is owned by the pylance team. I realize that you're not using pylance, since it works only with VS Code, but the auto-import logic is common to pylance and pyright. You could log a feature request in the [pylance-release project](https://github.com/microsoft/pylance-release).

**Reply by mezuzza** [reply]
Appreciate it. I'll submit an issue over there in a bit.

Just to align a bit more, do you also feel similarly if the module was a public one which reexported a value i.e. `_bar` was actually `bar`? I'm a bit conflicted on that one.

**Reply by erictraut** [reply]
Not sure. Yeah, there are tricky details that would need to be worked out.
